### PR TITLE
Fix named split sorting and remove unnecessary casting

### PR DESCRIPTION
**Pull request by albertvillanova**
This PR:
- makes `NamedSplit` sortable: so that `sorted()` can be called on them
- removes unnecessary `sorted()` on `dict.keys()`: `dict_keys` view is already like a `set`
- removes unnecessary casting of `NamedSplit` to `str`

### CODE REVIEW

File: src/datasets/data_files.py [hunk 1]
```diff
@@ -14,7 +14,7 @@
 from .utils.py_utils import string_to_dict
 
 
-DEFAULT_SPLIT = str(Split.TRAIN)
+SANITIZED_DEFAULT_SPLIT = str(Split.TRAIN)
```
Comment (lhoestq) [inline src/datasets/data_files.py:17]: (nit) I think you can keep DEFAULT_SPLIT and call str() in sanitize_patterns, as sanitize_patterns already calls str() to the users requested splits.

**Review by lhoestq — APPROVED** [review]
sounds good to me !

**Comment by albertvillanova** [comment]
Sorry, just too sync... :sweat_smile:

File: src/datasets/data_files.py (hunk 1 above)
Comment (albertvillanova) [reply src/datasets/data_files.py:17]: @lhoestq that makes sense if `DEFAULT_SPLIT` is used anywhere else than in `sanitize_patterns`, but this is not currently the case.

**Pull request merged**